lib/std/hash_map.zig

```diff
@@ -92,6 +92,34 @@ pub fn hashString(s: []const u8) u64 {
     return std.hash.Wyhash.hash(0, s);
 }
 
+pub const StringIndexContext = struct {
+    bytes: *std.ArrayListUnmanaged(u8),
+
+    pub fn eql(self: @This(), a: u32, b: u32) bool {
+        _ = self;
+        return a == b;
+    }
+
+    pub fn hash(self: @This(), x: u32) u64 {
+        const x_slice = mem.spanZ(@ptrCast([*:0]const u8, self.bytes.items.ptr) + x);
```

Comment:
Should this use `mem.sliceTo` instead? 

>  ```
>   /// Deprecated: use std.mem.span() or std.mem.sliceTo()
>   /// Same as `span`, except when there is both a sentinel and an array
>   /// length or slice length, scans the memory for the sentinel value
>   /// rather than using the length.
>   pub fn spanZ(ptr: anytype) Span(@TypeOf(ptr)) {
>   ```